girder/utility/cache.py

```diff
@@ -0,0 +1,28 @@
+import cherrypy
+from dogpile.cache import make_region, register_backend
+from dogpile.cache.backends.memory import MemoryBackend
+
+
+class CherrypyRequestBackend(MemoryBackend):
+    """
+    A memory backed cache for individual CherryPy requests.
+
+    This provides a cache backend for dogpile.cache which is designed
+    to work in a thread-safe manner using cherrypy.request, a thread local
+    storage that only lasts for the duration of a request.
+    """
+    def __init__(self, arguments):
+        pass
+
+    @property
+    def _cache(self):
```

Comment:
This method should probably check that `cherrypy.request` exists and raise an exception if the developer is trying to use the request cache outside of a request context.